clients/admin-ui/src/features/datamap/SpatialDatamap.tsx

```diff
@@ -34,26 +37,30 @@ const useSpatialDatamap = (rows: Row<DatamapRow>[]) => {
         }, {} as Record<string, SystemNode>),
     [rows]
   );
-
   const data = useMemo(() => {
     let nodes: SystemNode[] = [];
-    let links: Link[] = [];
+    const links: Set<Link> = new Set([]);
```

Comment:
I redid the link creation to use a set because it's possible that two systems could create identical links that would end up rendering on top of each other

example:

```
system_1.ingress = [system_2]
system_2.egress = [system_1]
```

Both of those would create an identical link object.

Reply:
woof this is tricky. I suspect the FE may need to dedupe these further to make the map usable. right now we get a lot of bidirectional arrows which aren't useful. for example, why do we have double arrows going back and forth from `api-service` to `profile-service`?

<img width="367" alt="image" src="https://user-images.githubusercontent.com/24641006/230985273-558b7d7a-d9cc-4a7d-9e03-d066739f9bf4.png">


as you mentioned we can do it in a follow up, though we should check with product if it's okay to merge this one for the release without that follow up. maybe it's something the backend can help with, I'm not sure without digging into the endpoint data more

Reply:
I'll look the deduping code once more so see if there is anything subtle. 

I confirmed with Rachel that it's okay if we add bi-directional links in the future that contain 1 line and 2 arrows as well as fixing the zoomed out graph. Currently no one has a really complex graph like this with a ton of links so it won't show up anywhere


https://github.com/ethyca/fides/issues/3030